### PR TITLE
Allow reading of config in sql backends

### LINKED ISSUE #1157: MySQL/PostgreSQL secret backend: Allow reading configuration back out
Currently the configuration endpoints for both of the SQL backends only support POST, not GET.

 * `/{mysql|postgresql}/config/connection`
 * `/{mysql|postgresql}/config/lease`

I'm attempting to write code that configures these backends via API, and I'd like to verify that my configuration makes it into Vault. Currently there's no way to do this other than functionally testing the effects of a given config.

Outside of testing, I think from an operational perspective it's useful to be able to inspect the live configuration of a backend. For example, it'd be important to know what `max_open_connections` is set to when looking for performance bottlenecks in a running system.

### PR DESCRIPTION
**Pull request by vishalnayak**
Fixes #1157 

May solve #317 as well.

### CODE REVIEW

**Comment by jefferai** [comment]
I believe the original reason that this wasn't allowed is because the configs contain passwords. That said, the ACL system can more than partition who has access...

File: builtin/logical/mysql/backend.go [hunk 1]
```diff
@@ -68,12 +68,7 @@ func (b *backend) DB(s logical.Storage) (*sql.DB, error) {
 		return nil, err
 	}
 
-	conn := connConfig.ConnectionURL
-	if len(conn) == 0 {
-		conn = connConfig.ConnectionString
-	}
-
-	b.db, err = sql.Open("mysql", conn)
+	b.db, err = sql.Open("mysql", connConfig.ConnectionURL)
```
Comment (jefferai) [inline builtin/logical/mysql/backend.go]: Why change this?

File: builtin/logical/mysql/path_config_connection.go [hunk 2]
```diff
@@ -17,12 +18,6 @@ func pathConfigConnection(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "DB connection string",
 			},
```
Comment (jefferai) [inline builtin/logical/mysql/path_config_connection.go]: Don't get rid of this. We'll have something newer soon enough anyways, and there's no point potentially breaking backwards compatibility.

File: builtin/logical/mysql/path_config_connection.go [hunk 3]
```diff
@@ -95,10 +106,9 @@ func (b *backend) pathConnectionWrite(
 }
 
 type connectionConfig struct {
```
Comment (jefferai) [inline builtin/logical/mysql/path_config_connection.go]: Same here, please restore the old value.

File: builtin/logical/mysql/backend.go (hunk 1 above)
Comment (vishalnayak) [inline builtin/logical/mysql/backend.go]: Done. Retained the old values everywhere.

**Comment by vishalnayak** [comment]
Got LGTM from Jefferai and LLBennett offline. Merging.

**Pull request merged**